core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -455,6 +455,7 @@ class GroupMetadataManager(brokerId: Int,
         loadGroupsAndOffsets(topicPartition, onGroupLoaded)
         info(s"Finished loading offsets and group metadata from $topicPartition in ${time.milliseconds() - startMs} milliseconds.")
       } catch {
+        case e: IOException => throw new KafkaStorageException(s"Error loading offsets from $topicPartition", e)
```

Comment:
Good catch! I missed the fact that this method is run in a scheduler. Sure. I have updated the patch to call RelicaManager.handleLogDirFailure() in `doLoadGroupsAndOffsets()` when there is `IOException`. I also reverted the change that was made to update the `responseMap` when there is `IOException`.

Becket had similar comment and asked me to do the same in `TransactionStateManager`. I didn't do that because I am not sure I know `TransactionStateManager` good enough to handle IOException in the best way. This is an existing problem with `TransactionStateManager` and probably other modules in Kafka as well because they don't explicitly handle the scenario when the disk write operation fails. Can we focus on the IOException that currently causes `halt()` and leave the handling of IOExeption that are currently ignored (e.g. in `TransactionStateManager`) in follow up patch?